common/src/main/java/dev/isxander/yacl3/gui/controllers/dropdown/DropdownStringControllerElement.java

```diff
@@ -0,0 +1,37 @@
+package dev.isxander.yacl3.gui.controllers.dropdown;
+
+import dev.isxander.yacl3.api.utils.Dimension;
+import dev.isxander.yacl3.gui.YACLScreen;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class DropdownStringControllerElement extends AbstractDropdownControllerElement<String, String> {
+	private final DropdownStringController controller;
+
+	public DropdownStringControllerElement(DropdownStringController control, YACLScreen screen, Dimension<Integer> dim) {
+		super(control, screen, dim);
+		this.controller = control;
+	}
+
+	@Override
+	public List<String> getMatchingValues() {
+		var stream = controller.getAllowedValues().stream();
+		if (controller.allowMode == DropdownMode.ALLOW_ANY && !controller.getAllowedValues().contains(inputField)) {
+			stream = Stream.concat(Stream.of(inputField), stream);
+		}
+		return stream
+				.filter(this::matchingValue)
+				.sorted((s1, s2) -> {
+					if (s1.startsWith(inputField) && !s2.startsWith(inputField)) return -1;
+					if (!s1.startsWith(inputField) && s2.startsWith(inputField)) return 1;
+					return s1.compareTo(s2);
+				})
+				.toList();
+	}
```

Comment:
Could this method be cached when the user modifies the input string? I think this would work: 
```java
@Override
public boolean modifyInput(Consumer<StringBuilder> builder) {
    boolean success = super.modifyInput(builder);
    if (success) {
        this.computeMatchingValues();
    }
    return success;
}
```

Reply:
I remember wanting to do this; I wonder why I never did. Thanks a lot for catching that.